libsolidity/interface/StandardCompiler.cpp

```diff
@@ -1164,8 +1180,102 @@ std::map<std::string, Json::Value> StandardCompiler::parseAstFromInput(StringMap
 	return sourceJsons;
 }
 
+Json::Value StandardCompiler::importEVMAssembly(StandardCompiler::InputsAndSettings _inputsAndSettings)
+{
+	solAssert(_inputsAndSettings.language == "EVMAssembly");
+	solAssert(_inputsAndSettings.sources.empty());
+	solAssert(_inputsAndSettings.jsonSources.size() == 1);
+
+	if (!isBinaryRequested(_inputsAndSettings.outputSelection))
+		return Json::objectValue;
+
+	evmasm::EVMAssemblyStack stack(_inputsAndSettings.evmVersion);
+	std::string const& sourceName = _inputsAndSettings.jsonSources.begin()->first; // result of structured binding can only be used within lambda from C++20 on.
+	Json::Value const& sourceJson = _inputsAndSettings.jsonSources.begin()->second;
+	try
+	{
+		stack.analyze(sourceName, sourceJson);
+		stack.assemble();
+	}
+	catch (util::Exception const& e)
+	{
+		return formatFatalError(Error::Type::Exception, "Exception during assembly import: " + std::string(e.what()));
+	}
```

Comment:
Unlike AST import, asm import actually has proper validations for most things, so you should just catch specifically that validation error here (`AssemblyImportException`). I don't remember any other exceptions it might throw, but if there are any, they should be caught separately too.

And it needs a general `(...)` handler for unexpected stuff, like assertions, because all of these must still be reported inside Standard JSON.


```suggestion
	catch (AssemblyImportException const& e)
	{
		return formatFatalError(Error::Type::Exception, "Assembly import error: " + std::string(e.what()));
	}
	catch (...)
	{
		errors.append(formatError(
			Error::Type::Exception,
			"general",
			"Unknown exception during assembly import: " + boost::current_exception_diagnostic_information()
		));
	}
```

Reply:
e.g. `InvalidOpcode ` is defined as `DEV_SIMPLE_EXCEPTION(InvalidOpcode);` that is `util::Exception`, because  `AssemblyImportException` is also a `util::Exception`, I will just leave it how it was.

Reply:
however, we decided in the chat that we will still catch also `evmasm::AssemblyImportException`